.github/workflows/stable.yml

```diff
@@ -131,8 +162,22 @@ jobs:
         run: tools/ci/submit_coverage.sh
         if: ${{ always() }}
       - name: Upload pytest test results
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
         with:
           name: pytest-results-${{ matrix.os }}-${{ matrix.python-version }}
           path: for_testing/test-results.xml
         if: ${{ always() && matrix.check == 'test' }}
+
+  publish:
```

Comment:
The reason that I want to do this is because with 4.0.0, there were tests that passed, then I tagged, and forgot to test on the tagged version before publishing. `DeprecationWarning`s mutated into `ExpiredDeprecationError`s and we had a release that failed its own tests.

That said, it [looks like](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments) I can create an environment that requires manual review before providing access to the PYPI token. Would that feel more comfortable?

Reply:
Honesty - your call - you're doing the releases.